### PR TITLE
feat(offer): multi currency

### PR DESCRIPTION
**Pull request by nduchak**
integrate offer creating with multi-currency

### CODE REVIEW

File: src/context/Services/storage/interfaces.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import { Dispatch } from 'react'
+import network from '../../../blockchain/config'
```
Comment (farinalvaro) [inline src/context/Services/storage/interfaces.ts:2]: This made me realize of [this issue](https://github.com/rsksmart/rif-marketplace-ui/issues/360) 🤦‍♂️. For now, could you please use `import networkConfig from 'config'`?

Comment (farinalvaro) [inline src/context/Services/storage/interfaces.ts:2]: This made me realize of [this issue](https://github.com/rsksmart/rif-marketplace-ui/issues/360) 🤦‍♂️. For now may be use `import networkConfig from 'config'`? What do you think @jurajpiar?
Also, all paths could be relative to `src` folder

File: src/components/pages/storage/sell/StorageSellPage.tsx [hunk 2]
```diff
@@ -54,31 +54,36 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+
+          if (tokenIndex !== -1) {
+            acc.periods[tokenIndex].push(timePeriod * PeriodInSeconds.Daily)
+            acc.prices[tokenIndex].push(weiPrice)
+            return acc
+          }
+
+          return {
+            prices: [...acc.prices, [weiPrice]],
+            periods: [...acc.periods, [timePeriod * PeriodInSeconds.Daily]],
```
Comment (farinalvaro) [inline src/components/pages/storage/sell/StorageSellPage.tsx:80]: Do we need the second item as an array here or could we simply do `[...acc.prices, weiPrice]`?

**Review by farinalvaro — COMMENTED** [review]
Looking good 🙂 just pointed some minor things.
Also, please fix the linter errors

File: src/context/Services/storage/interfaces.ts (hunk 1 above)
Comment (nduchak) [reply src/context/Services/storage/interfaces.ts]: Thanks for your comment.
Done

File: src/components/pages/storage/sell/StorageSellPage.tsx (hunk 2 above)
Comment (nduchak) [reply src/components/pages/storage/sell/StorageSellPage.tsx:80]: Actually no, we should have a structure like `{ prices: string[][], periods: number[][], tokens: string[] }`
where each of tokens have price and period array on specific index

**Review by farinalvaro — APPROVED** [review]
LGTM :)

File: src/context/Services/storage/interfaces.ts [hunk 3]
```diff
@@ -51,3 +53,10 @@ export enum TimePeriodEnum {
 }
 
 export type STORAGE_ACTION = STORAGE_SELL_ACTION | OFFERS_ACTION
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export const TOKENS_ADDRESSES = {
```
Comment (jurajpiar) [inline src/context/Services/storage/interfaces.ts:59]: good idea, just rename it to TOKEN_ADDRESSES

File: src/contracts/Storage.ts [hunk 4]
```diff
@@ -57,12 +58,12 @@ class StorageContract {
...
 
     const estimatedGas = await this.contract.methods
-      .setOffer(capacityMB, billingPeriods, billingRbtcWeiPrices, prefixedMsg)
+      .setOffer(capacityMB, billingPeriods, billingRbtcWeiPrices, tokens, prefixedMsg)
       .estimateGas({ from, gasPrice })
     const gas = Math.floor(estimatedGas * 1.1)
 
     return this.contract.methods
-      .setOffer(capacityMB, billingPeriods, billingRbtcWeiPrices, prefixedMsg)
+      .setOffer(capacityMB, billingPeriods, billingRbtcWeiPrices, tokens, prefixedMsg)
```
Comment (jurajpiar) [inline src/contracts/Storage.ts:66]: I'd split this so to:
```
const setOfferTask = this.contract.methods.setOffer(capacityMB, billingPeriods, billingRbtcWeiPrices, tokens, prefixedMsg)
```
and only then use that to both estimate gas and send tx

File: src/context/Services/storage/interfaces.ts (hunk 3 above)
Comment (jurajpiar) [reply src/context/Services/storage/interfaces.ts:59]: also not sure it should be capitalised as its value is only known in runtime.

File: src/components/pages/storage/sell/StorageSellPage.tsx [hunk 5]
```diff
@@ -54,31 +54,36 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+        return acc
+      }
+
+      return {
+        prices: [...acc.prices, [weiPrice]],
+        periods: [...acc.periods, [timePeriod * PeriodInSeconds.Daily]],
+        tokens: [...acc.tokens, TOKENS_ADDRESSES[currency]],
+      }
+    },
+        { prices: [], periods: [], tokens: [] } as any,
```
Comment (jurajpiar) [inline src/components/pages/storage/sell/StorageSellPage.tsx:84]: did you have to cast to any here? Would `.reduce<{ prices: [], periods: [], tokens: [] }>(...` make a difference?

File: src/contracts/Storage.ts (hunk 4 above)
Comment (nduchak) [reply src/contracts/Storage.ts]: Done

File: src/context/Services/storage/interfaces.ts [hunk 6]
```diff
@@ -51,3 +53,10 @@ export enum TimePeriodEnum {
 }
 
 export type STORAGE_ACTION = STORAGE_SELL_ACTION | OFFERS_ACTION
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export const TokenAddressees = {
```
Comment (jurajpiar) [inline src/context/Services/storage/interfaces.ts:59]: haha sorry man, now it starts with capital letter...

Comment (nduchak) [reply src/context/Services/storage/interfaces.ts:59]: Yea, i just fix it

File: src/components/pages/storage/sell/StorageSellPage.tsx (hunk 5 above)
Comment (nduchak) [reply src/components/pages/storage/sell/StorageSellPage.tsx:84]: nope, it's doesn't make a difference

Comment (jurajpiar) [reply src/components/pages/storage/sell/StorageSellPage.tsx:84]: ah ok.

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**